### ISSUE TITLE
Bluetooth connection to open car from Watch 

### ISSUE THREAD

**Issue by menocode** [Kemmey/Kemmey-TeslaWatch-Public#524]
I tried the new feature to connect my watch via bluetooth and open the car.

I was able to connect "something' to my car using "Enable Bluetooth direct vehicle control" on my iphone. Not sure if this connected my iphone or my watch as an additional key. I did see an additional key in the Tesla, but walking towards car with Watch did not open car and I don't see a bluetooth connection between in the watch connected to the car. 

Also, don't see any notification in Watch. 

I tried to connect Bluetooth in car with watch - car did not find watch and vice versa. Running latest version of watch OS and latest version of Tesla Software.

**Comment by deevodavis71** [comment]
Exact same results for me as per the comments above. Not sure what is new about this? Was hoping that just the watch would be enough to open the car, but if I need my phone too then it's no different to what I already had.

**Comment by ChrisFuchi** [comment]
Same issues here. From watch on my iPhone I uninstalled and reinstalled the Tesla watch app and my watch now displays  "Enable Bluetooth direct"  I did add a key as well but it displays as an undefined key. When I try to add a new device it is not recognized even after reboots of my car and apple watch. I do have BT functionality from the apple watch with no iPhone but only manually. When I walk up to the car it doesn't unlock automatically and I also got an in car message saying walk away door locking is disabled and lock manually.

**Comment by onielmartinjr** [comment]
I am in the exact same position. I went through the pairing process in my Model Y, I placed the key card on the console, the car then asked to confirm the new key pairing, I confirmed it, and then my Tesla App notified me a new key card was added. If I open the Tesla Watch App on my iPhone, I see the following:

![1e44738e-64ae-4b35-9e42-c2b2f808e0f2](https://user-images.githubusercontent.com/13435470/209984948-0d3d9089-f5ec-4a93-8444-b98d9f98700d.jpg)

But when I approach my Model Y with the **Watch for Tesla App** actively running, the door handle does not open for me. I also do not see a Bluetooth icon on the watch or anything that indicates my watch is authenticating me to the car. I have tried resetting my car, my watch, my phone, also re-pairing the key card over 5 times. I do see a key card on the screen added, but it looks like the new key card never successfully connects.

**Comment by Kemmey** [comment]
Hey everyone - you need to do this ON YOUR WATCHES. If you pair your phone as key, your phone will work as key - not your watch. You need to pair your watch - so do exactly what you've all done, but using the app on the watch. 

Then it will work, when the app is in the foreground on the watch. Due to battery management, the app has to be open for Bluetooth to be available.

**Comment by onielmartinjr** [comment]
> Hey everyone - you need to do this ON YOUR WATCHES. If you pair your phone as key, your phone will work as key - not your watch. You need to pair your watch - so do exactly what you've all done, but using the app on the watch.
> 
> Then it will work, when the app is in the foreground on the watch. Due to battery management, the app has to be open for Bluetooth to be available.

Confirmed this solved the issue for me. I kinda feel like an idiot for not realizing myself sooner. Thanks for the quick reply and the great app!

**Comment by Kemmey** [comment]
I'll reword some things to make this more obvious in a future update - A LOT of people have fallen into this trap.

**Comment by cosmostail** [comment]
Is there a workaround to leave watch app always running? I don’t like the idea that I need to tap the app open before open the car door.

**Comment by Pinasan** [comment]
I have tried this feature numerous time on my watch. It pairs initially, but when I walk away, it needs to be paired again. Any suggestions on what I may be doing wrong? Thanks!

**Comment by Kemmey** [comment]
> @Kemmey   Is there a workaround to leave watch app always running? I don’t like the idea that I need to tap the app open before open the car door. 

No - it's a battery management issue due to the small battery in the watch.

**Comment by Kemmey** [comment]
> I have tried this feature numerous time on my watch. It pairs initially, but when I walk away, it needs to be paired again. Any suggestions on what I may be doing wrong? Thanks! 

Could you send me the debug logs? You can find them in the app under settings at the bottom. Logs syncs between devices, so you can send them from your phone.

**Comment by yuriw** [comment]
I can open my MY only after I open an app on my watch :(

**Comment by Kemmey** [comment]
> I can open my MY only after I open an app on my watch :(

But you can now do so with zero delay waiting for the car to wake and get online. And you can do so with a non-lte watch, and you can do it in an underground parking garage or in the middle of a desert or deep forest. 

You're welcome :-D

**Comment by yuriw** [comment]
@Kemmey are you suggesting that this is NOT an expected behavior?

**Comment by Kemmey** [comment]
> @Kemmey are you suggesting that this is NOT an expected behavior?
> 
> 
> 
>  

No, I'm saying your experience is exactly the expected behavior. Your watch has an extremely tiny battery, so due to battery management, only the foreground app will have Bluetooth connectivity privileges. This is managed by WatchOS. You need to open the app on your watch, wait for Bluetooth connection (which is usually instant) and then just grab a door handle, get in, put in drive and go. 

Not sure how you can be disappointed by that, compared to the old model of waiting up to two minutes for the car to wake, when both watch and car has connectivity. If those conditions are not satisfied, it wouldn't work AT ALL.

**Comment by yuriw** [comment]
@Kemmey I did not say I was disappointed, was just wondering if it's expected, thx for the clarification

**Comment by Kemmey** [comment]
> @Kemmey I did not say I was disappointed, was just wondering if it's expected, thx for the clarification

Sorry for being on the defense - getting a lot of flack for this awesome feature :-D

**Comment by ChrisFuchi** [comment]
I'm one of the introductory purchasers of this app few years ago and even without BT it has always been my favourite app to use with my apple watch especially when compared to some other apps with apple watch complications that don;t work well at all.  Thank you for this and your fast response and support!

**Comment by Pinasan** [comment]
Hi Kim,
I know this wasn’t directed to me or my issue, but I love the app… even if
I haven’t been able to get it to work. Yet!

Thanks!

David

**Comment by boil42** [comment]
May not be totally convenient for all usage but for me "brilliant" for what I go out to the car for something and leave my phone inside the house :)

**Comment by WilleyMunda** [comment]
I have added a key via my watch for Tesla app and it shows in the car as a digital key called Apple Watch. However I wanted to test opening the vehicle like if I had no cellular connection and as my watch has cellular I disable that and also put my watch on flight mode so that I could see if I could open and drive my car with no phone or no signal to my watch but it just says no network connection on the app on my phone? Should my watch be able to open the car if I only have my watch with no signal and no phone or Tesla card key?

**Comment by WilleyMunda** [comment]
> Hey everyone - you need to do this ON YOUR WATCHES. If you pair your phone as key, your phone will work as key - not your watch. You need to pair your watch - so do exactly what you've all done, but using the app on the watch.
> 


I did it on the phone app and also the watch app are you suggesting to remove the one I've done for the watch for Tesla app on the phone?

**Comment by menocode** [comment]
I initially used the app when the first version of this came out last week. The app on the watch did not initially include the option " Enable Bluetooth..." 
Since then I have tried to pair directly from the watch app to the car 3 times (including the version that dropped yesterday. The car shows a new key.
- However, I need to name that key in the car app 
- The Watch app does not show that key
- If I'm away from the car, watch app says" Vehicle not in range" . If I'm close to the car, it says "pair with car"
It seems that the pairing is not happening correctly. What might I be doing wrong.

**Comment by Kemmey** [comment]
Please share version and build number of the app (visible in settings at top)

Try to do the pairing, close the app, rename in car under keys, then open app again. Check the debug logs - there's a Bluetooth heading - you can share that from your phone via mail to me - logs from the Watch sync to the phone - you can see origin device in the logs.

**Comment by Kemmey** [comment]
And anecdotally - this has been working perfectly on all my devices and model 3 and model x for past few weeks. But I do also have reports of users where renaming keys loses the pairing. I've not been able to replicate that, so will have to do more investigating to figure out. 

Same applies to walk away lock - works perfectly for me - others, not so much. Same vehicle, same software. So weird.

**Comment by Pinasan** [comment]
Hi Kim,
I was losing the pairing of the key on my watch. The solution was to remove
all the keys paired with my car and add them back (I had some old keys that
I no longer used). It has worked flawlessly since!

Thanks!

David

**Comment by dapper27** [comment]
> Same applies to walk away lock - works perfectly for me - others, not so much. Same vehicle, same software. So weird. 

Does the watch app need to be active for walk away to work? Also, if the app is active but the screen goes to sleep, is Bluetooth still active?

**Comment by Kemmey** [comment]
If app goes to background, Bluetooth turns off and walk away triggers immediately. For me. 

Watch screen can turn off - but app is only then active for a configurable time of between 30 and 70 seconds

**Comment by ChrisFuchi** [comment]
So tonight I connected my apple watch as a key from the watch. Said to tap my card and it added the key. I did not add key from my iPhone or from the cars add key button. I went inside my house, turned off my iPhone. Went back out to car. I opened the app and was able to unlock and open the car just using my apple watch. I also tried the unlatch feature which allows you to unlatch any door. I walked away from the car and went into my house and heard my Tesla auto lock. Success!

**Comment by ClementKMLi** [comment]
may I ask if I need to have the phone connected to pair the watch key?? as I've tried to pair my watch a few times (with my phone turns to airplane mode), it does appear a key on the car but that key doesn't seem work at all

**Comment by Kemmey** [comment]
@ClementKMLi you can leave the phone in your home when setting this up. The watch works completely independently via Bluetooth. 

You do need to go through the pairing process on the watch. Once done, your watch will work as a key, as long as the app is running in the foreground on the watch. So you do need to open the app on the watch for it to work - but when it is active, just grab the door handle and press brake as you normally would.

**Comment by ClementKMLi** [comment]
> @ClementKMLi you can leave the phone in your home when setting this up. The watch works completely independently via Bluetooth.
> 
> You do need to go through the pairing process on the watch. Once done, your watch will work as a key, as long as the app is running in the foreground on the watch. So you do need to open the app on the watch for it to work - but when it is active, just grab the door handle and press brake as you normally would.

Thanks Kemmey, may I ask if I need to first pair the watch bluetooth with the car?? or does it pair up automatically?? I am sorry it's been a while since I've setup my phone key, I forgot how to do it....

**Comment by Kemmey** [comment]
Yeah - you need to pair the watch as key. Bring your keycard to the car, open the app on your watch, tap settings, enable Bluetooth direct control, wait for it to find your car, tap add key, tap your keycard to the console in the car, tap confirm on the car screen - and done :-D

**Comment by ClementKMLi** [comment]
Hi Kenneth,

Thanks, Ive tried, it reads a new key is added, but it doesn’t seems to be working, it shows a key is added, but when I take away my key card, it just couldn’t start….. nor I can open frunk/trunk etc….

Would you be able to help me out?

Best Regards with Blessings

Clement Li
Sent from my mobile device

On 10 Jan 2023, at 17:56, Kim Hansen ***@***.***> wrote:

﻿

Yeah - you need to pair the watch as key. Bring your keycard to the car, open the app on your watch, tap settings, enable Bluetooth direct control, wait for it to find your car, tap add key, tap your keycard to the console in the car, tap confirm on the car screen - and done :-D

**Comment by Harry-Butler** [comment]
Hi,

Just a follow up question. Bluetooth key on watch all working fine on Model Y... Open car, start driving, no problem. Once driving and watch app closes, the car pops up a warning "key not detected" I assume this is fine? It just means you will need to open the app again to start another driving session. Is that correct? Cheers and nice work with the App.
